bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/util/SolutionProvider.scala

```diff
@@ -25,9 +27,12 @@ object SolutionProvider {
     Site.GS -> new SolutionProvider(Site.GS)
   )
 
-  def apply(site: Site) = providers(site)
-  def apply(peer: Peer) = providers(peer.site)
-  def apply(ctx: QvContext) = providers(ctx.site)
+  def currentNight(site: Site, currentTime: Long): Option[Night] =
+    SemesterData.current(site, currentTime).nights.find(n => n.dayEnd > currentTime)
```

Comment:
This is the core change of this PR it used to be `n.end > currentTime` where `n.end` is the sunrise time